app/services/update_saved_scenario.rb

```diff
@@ -0,0 +1,62 @@
+# frozen_string_literal: true
+
+# Creates a new API scenario based on the given existing API `scenario_id`, and
+# then adds it to the provided SavedScenario adding the old scenario to history.
+# The new API scenario will be marked as protected.
+#
+# saved_scenario  - The scenario to be updated
+# scenario_id     - The ID of the scenario to be saved.
+# settings        - Optional extra scenario data to be sent to ETEngine when
+#                   creating the new API scenario.
+#
+# Returns a ServiceResult with the resulting SavedScenario.
+class UpdateSavedScenario
+  attr_reader :saved_scenario, :scenario_id, :settings
+
+  # Make the class act like a lambda
+  def self.call *args, &block
+    new.call *args, &block
+  end
+
+  # Make an instance act like a lambda
+  def call(saved_scenario, scenario_id, settings = {})
+    @saved_scenario, @scenario_id, @settings =
+      saved_scenario, scenario_id, settings
+
+    old_scenario = saved_scenario.scenario
+    return api_response if failure?
+
+    saved_scenario.tap do |ss|
+      ss.scenario_id_history << ss.scenario_id
+      ss.scenario_id = api_scenario.id
+      ss.title = old_scenario.title
+      unprotect and return failure unless saved_scenario.valid?
+      ss.save
+      saved_scenario.scenario = api_scenario
+    end
+
+    ServiceResult.success(saved_scenario)
+  end
+
+  private
+
+  def unprotect
+    UnprotectAPIScenario.call api_scenario.id
+  end
+
+  def failure
+    ServiceResult.failure saved_scenario.errors.full_messages
+  end
+
+  def api_scenario
+    api_response.value
+  end
+
+  def api_response
+    @_resp ||= CreateAPIScenario.call settings.merge(scenario_id: scenario_id)
```

Comment:
I think the scenario title needs to be included here, otherwise it defaults to "API". This means that when you use the "Save scenario" button, the scenario appears to be renamed:

![Screen Shot 2019-06-18 at 12 07 37](https://user-images.githubusercontent.com/4383/59677431-bef5b480-91c1-11e9-964d-cfffd392b9e7.png)